test/unit/dynamic-variables.test.js

```diff
@@ -0,0 +1,111 @@
+var _ = require('lodash'),
+    dynamicVariables = require('../../lib/superstring/dynamic-variables'),
+    expect = require('chai').expect;
+
+describe('Dynamic variable', function () {
+    it('should have all attributes', function () {
+        _.forOwn(dynamicVariables, function (dynamicVariable) {
+            expect(dynamicVariable).to.be.an('object');
+            expect(dynamicVariable.description).to.be.a('string');
+            expect(dynamicVariable.generator).to.be.a('function');
+        });
+    });
+
+    it('should not be a duplicate', function () {
+        var dynamicVariableSet = [];
+
+        _.forOwn(dynamicVariables, function (dynamicVariable, name) {
+            expect(dynamicVariableSet[name]).to.be.undefined;
+            dynamicVariableSet[name] = dynamicVariable;
+        });
+    });
+
+
+    describe('generator', function () {
+        it('should return random data', function () {
+            _.forOwn(dynamicVariables, function (variable) {
+                expect(variable.generator()).to.not.be.undefined;
+            });
+        });
+    });
+
+    describe('$randomInt', function () {
```

Comment:
move all the `$*` describe blocks inside `generator` as an `it` block (an example is given above).